apollo-router/src/services/connector_service.rs

```diff
@@ -253,19 +259,55 @@ impl ConnectorServiceFactory {
         connector_request_service_factory: Arc<ConnectorRequestServiceFactory>,
         plugins: Arc<Plugins>,
     ) -> Self {
+        // Build connector services for each service
+        let mut services_map = HashMap::with_capacity(connectors_by_service_name.len());
+
+        for (connector_internal_name, connector) in connectors_by_service_name.iter() {
+            // Create the base connector service
+            let base_service = ConnectorService {
+                _schema: schema.clone(),
+                _subgraph_schemas: subgraph_schemas.clone(),
+                _subscription_config: subscription_config.clone(),
+                connectors_by_service_name: connectors_by_service_name.clone(),
+                connector_request_service_factory: connector_request_service_factory.clone(),
+            };
+            let subgraph_name = connector.id.subgraph_name.as_ref();
+            let source_name = connector.source_config_key();
+
+            // Apply plugins with the correct service name
+            let service_with_plugins =
+                plugins
+                    .iter()
+                    .rev()
+                    .fold(base_service.boxed(), |acc, (_, plugin)| {
+                        plugin.connector_service(
+                            subgraph_name,
+                            &source_name,
+                            connector_internal_name,
+                            acc,
+                        )
+                    });
+
+            // Buffer the service
+            let buffered_service = Buffer::new(service_with_plugins, DEFAULT_BUFFER_SIZE);
```

Comment:
There's an extension on Service which lets you use `servic_with_plugins.buffered()` IIRC and it will use default settings